packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/ModelFactoryProvider.cs

```diff
@@ -126,10 +126,11 @@ protected override MethodProvider[] BuildMethods()
                     parameters,
                     returns: new XmlDocReturnsStatement($"A new {modelProvider.Type:C} instance for mocking."));
 
+                var collectionInitialization = GetCollectionInitialization(signature);
                 var statements = new MethodBodyStatements(
                 [
-                    .. GetCollectionInitialization(signature),
-                    MethodBodyStatement.EmptyLine,
+                    .. collectionInitialization,
+                    collectionInitialization.Count > 0 ? MethodBodyStatement.EmptyLine : MethodBodyStatement.Empty,
                     Return(New.Instance(typeToInstantiate.Type, [.. GetCtorArgs(modelProvider, signature, fullConstructor, binaryDataParam)]))
                 ]);
```

Comment:
do you think if it is possible that we make something when `MethodBodyStatements` writes itself to ignore leading and trailing empty statements, and maybe consecutive empty statements?

Reply:
in the meantime I am fine with the changes here

Reply:
I think there could be cases where you have two sets of MethodBodyStatements in which you would want the empty line in the second one. I think it is better to be explicit.